### PR TITLE
Add support for cluster delete-after annotation.

### PR DESCRIPTION
**Pull request by dgoodwin**
Specifies a duration to be added to creation time, after which the
cluster deployment is considered expired and should be cleaned up.
Cluster deployments will be requeued for one minute after this time to
ensure timely cleanup.

### CODE REVIEW

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 1]
```diff
@@ -467,6 +470,31 @@ func (c *Controller) syncClusterDeployment(key string) error {
...
+			expiry := clusterDeployment.CreationTimestamp.Add(dur)
+			clusterDeploymentLog.Debugf("cluster expires at: %s", expiry)
+			if time.Now().After(expiry) {
+				clusterDeploymentLog.WithField("expiry", expiry).Info("cluster has expired, issuing delete")
+				c.clustopClient.ClusteroperatorV1alpha1().ClusterDeployments(clusterDeployment.Namespace).Delete(clusterDeployment.Name, &metav1.DeleteOptions{})
+				return nil
+			}
+
+			// We have an expiry time, we're not expired yet, lets make sure to enqueue the cluster for
+			// just after it's expiry time:
```
Comment (csrwng) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go]: s/it's/its/

File: contrib/examples/cluster-deployment-template.yaml [hunk 2]
```diff
@@ -84,6 +84,8 @@ objects:
   metadata:
     name: ${CLUSTER_NAME}
     namespace: ${CLUSTER_NAMESPACE}
+    annotations:
+      clusteroperator.openshift.io/delete-after: 5m
```
Comment (csrwng) [inline contrib/examples/cluster-deployment-template.yaml]: Why so low? Right now it takes at least 20m for the cluster to finish provisioning

Comment (dgoodwin) [reply contrib/examples/cluster-deployment-template.yaml]: My mistake. I'm going to probably remove this in the default template, it'll bite someone no matter what we put in there.

**Comment by dgoodwin** [comment]
Fixed.

**Pull request merged**